tensorflow/lite/micro/micro_allocator.cc

```diff
@@ -116,6 +118,34 @@ TfLiteStatus CommitPlan(ErrorReporter* error_reporter,
   return kTfLiteOk;
 }
 
+IPersistentBufferAllocator* CreatePersistentArenaAllocator(uint8_t* buffer_head,
+                                                           size_t buffer_size) {
+  PersistentArenaBufferAllocator tmp =
+      PersistentArenaBufferAllocator(buffer_head, buffer_size);
+
+  // Allocate enough bytes from the buffer to create a SimpleMemoryAllocator.
+  // The new instance will use the current adjusted tail buffer from the tmp
+  // allocator instance.
+  uint8_t* allocator_buffer =
+      tmp.AllocatePersistentBuffer(sizeof(PersistentArenaBufferAllocator),
+                                   alignof(PersistentArenaBufferAllocator));
+  // Use the default copy constructor to populate internal states.
+  return new (allocator_buffer) PersistentArenaBufferAllocator(tmp);
+}
+
+INonPersistentBufferAllocator* CreateNonPersistentArenaAllocator(
+    uint8_t* buffer_head, size_t buffer_size,
+    IPersistentBufferAllocator* persistent_buffer_allocator) {
+  uint8_t* allocator_buffer =
+      persistent_buffer_allocator->AllocatePersistentBuffer(
```

Comment:
It might help avoid confusion if a comment could be added explicitly documenting that the NON-persistent arena buffer allocator object is actually created in a PERSISTENT buffer.